test_util/azure.py

```diff
@@ -35,77 +38,126 @@ def validate_hostname_prefix(prefix):
-                    log.error('{}: {}'.format(details.code, details.message))
-                error_msgs = '\n'.join(['{}: {}'.format(d.code, d.message) for d in result.error.details])
-                raise Exception("Template verification failed!\n{}".format(error_msgs))
+                log.critical('{}: {}'.format(result.error.code, result.error.message))
+                raise Exception("Template verification failed!\n{}".format(result.error))
             log.info('Template successfully validated')
             log.info('Starting template deployment')
-            azure_wrapper.rmc.deployments.create_or_update(
+            self.rmc.deployments.create_or_update(
                 group_name, deployment_name, deployment_properties)
             stack.pop_all()
+        log.info('Successfully started template deployment')
+
+    def create_deployment_properties(self, template_url, parameters):
+        """ Pulls the targeted template, checks parameter specs and casts
+        user provided parameters to the appropriate type. Assertion is raised
+        if there are unused parameters
+        """
+        user_parameters = copy.copy(parameters)
+        type_cast_map = {
+            'string': str,
+            'secureString': str,
+            'int': int,
+            'bool': bool,
+            'object': check_json_object,
+            'secureObject': check_json_object,
+            'array': check_array}
+        log.debug('Pulling Azure template for parameter validation...')
+        r = requests.get(template_url)
+        r.raise_for_status()
+        template = r.json()
+        if 'parameters' not in template:
+            assert user_parameters is None, 'This template does not support parameters, ' \
+                'yet parameters were supplied: {}'.format(user_parameters)
+        log.debug('Constructing DeploymentProperties from user parameters: {}'.format(parameters))
+        template_parameters = {}
+        for k, v in template['parameters'].items():
+            if k in user_parameters:
+                # All templates parameters are required to have a type field.
+                # Azure requires that parameters be provided as {key: {'value': value}}.
+                template_parameters[k] = {
+                    'value': type_cast_map[v['type']](user_parameters.pop(k))}
+        log.debug('Final template parameters: {}'.format(template_parameters))
+        if len(user_parameters) > 0:
+            raise Exception('Unrecognized template parameters were supplied: {}'.format(user_parameters))
+        return DeploymentProperties(
+            template=template,
+            mode=DeploymentMode.incremental,
+            parameters=template_parameters)
+
+
+class DcosAzureResourceGroup:
+    def __init__(self, group_name, azure_wrapper):
+        self.group_name = group_name
+        self.azure_wrapper = azure_wrapper
+
+    @classmethod
+    def deploy_acs_template(
+            cls, azure_wrapper, template_url, group_name,
```

Comment:
Sure thing, however I will not spend too much time on describing the parameters, as we do not want to be hard-coding those anywhere in the long run. Once dcos-launch is ready, methods like this will be purged in favor of having config files in CI where we can tweak the parameters whenever